app/services/notify_service.rb

```diff
@@ -43,10 +43,19 @@ def blocked?
     blocked ||= (@notification.from_account.silenced? && !@recipient.following?(@notification.from_account))                       # Hellban
     blocked ||= (@recipient.user.settings.interactions['must_be_follower']  && !@notification.from_account.following?(@recipient)) # Options
     blocked ||= (@recipient.user.settings.interactions['must_be_following'] && !@recipient.following?(@notification.from_account)) # Options
+    blocked ||= conversation_muted?
     blocked ||= send("blocked_#{@notification.type}?")                                                                             # Type-dependent filters
     blocked
   end
 
+  def conversation_muted?
+    if @notification.target_status
+      !@notification.target_status.conversation_id.nil? && ConversationMute.where(account: @recipient, conversation_id: @notification.target_status.conversation_id).exists?
```

Comment:
Might put this into something like `@recipient.muting_conversation?(@notification.target_status.conversation)`, and have it handle nils itself...